### DISCUSSION TITLE
Help with multi monitor setup

### DISCUSSION THREAD

**Discussion by martines3000** [i3/i3#5076] [Q&A]
System:
- Arch Linux
- KDE
- i3
- Laptop + External monitor (connected as "display port over usbc "if it matters)

I don't have any major problems and everything so far works out of the box. The only issue I have (it's probably just some config setting I can't seem to find) is that when my system logs out (because of inactivity) or when I turn of the external monitor, all workspaces jump to the laptop monitor. 

The behavior I want is that they stay on that external monitor, so when I get back to my PC and log in, I don't need to rearrange everything again.

Is there a configuration option to disable automatic rearranging ? Or do I need to change something in `xrandr` config or somewhere else.

Thank you for the help.

**Comment by dwedia** [comment]
I am running a setup, where I have my laptop plugged in to a dock with 2 external monitors connected.
When I am connected to my dock I want to workspaces to be on specific monitors.

To do this, I use the following snippit in my i3 config file

#Set workspaces to specific monitors
workspace 1 output DisplayPort-2 
workspace 2 output DisplayPort-2 
workspace 3 output DisplayPort-4
workspace 4 output DisplayPort-4
workspace 5 output DisplayPort-4
workspace 6 output DisplayPort-4
workspace 7 output DisplayPort-4
workspace 8 output DisplayPort-4 eDP
workspace 9 output DisplayPort-4 eDP
workspace 0 output DisplayPort-4 eDP

the eDP monitor is the laptops internal display.

This is working most of the time (I sometimes have to switch to the correct workplaces on the monitors I want, when I first turn the machine on. but this seems to persist well enough through screen locks
